crates/admin/src/metadata_api/mod.rs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::{
+    collections::HashSet,
+    str::FromStr,
+    sync::{Arc, LazyLock},
+};
+
+use axum::{
+    body::Bytes,
+    extract::{Path, State},
+    response::{AppendHeaders, IntoResponse, Response},
+    routing::{delete, get, head, put},
+    Router,
+};
+use bytestring::ByteString;
+use http::{header::ToStrError, HeaderMap, StatusCode};
+
+use restate_core::metadata_store::{MetadataStore, VersionedValue};
+use restate_metadata_store::{MetadataStoreClient, Precondition, ReadError, WriteError};
+use restate_types::{metadata_store::keys, Version};
+
+/// ETag header.
+const HEADER_ETAG: &str = "ETag";
+
+/// If-Match header. Possible values are
+/// - none | *: no precondition
+/// - does-not-exist: key must not exist
+/// - <version>: Match given etag number
```

Comment:
Hmm, seeing these two side by side makes me think `If-Match: none` might be misconstrued for `does-not-exist`. The etag alternative of `If-None-Match` makes that more explicit. Maybe we should at least take out the `none` option for now if you agree?

Reply:
Yeah, i see what you mean.